src/systems/filecoin_vm/runtime/runtime.id

```diff
@@ -2,28 +2,24 @@ import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
 import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
 import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
 import exitcode "github.com/filecoin-project/specs/systems/filecoin_vm/runtime/exitcode"
-import filcrypto "github.com/filecoin-project/specs/algorithms/crypto"
 import indices "github.com/filecoin-project/specs/systems/filecoin_vm/indices"
 import ipld "github.com/filecoin-project/specs/libraries/ipld"
 import msg "github.com/filecoin-project/specs/systems/filecoin_vm/message"
+import chain "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/chain"
 
 // Runtime is the VM's internal runtime object.
 // this is everything that is accessible to actors, beyond parameters.
 type Runtime interface {
-    CurrEpoch() block.ChainEpoch
+    CurrEpoch()        block.ChainEpoch
 
-    // Randomness returns a (pseudo)random string for the given epoch and tag.
-    Randomness(tag filcrypto.DomainSeparationTag, epoch block.ChainEpoch) util.Randomness
-    RandomnessWithAuxSeed(
-        tag      filcrypto.DomainSeparationTag
-        epoch    block.ChainEpoch
-        auxSeed  util.Serialization
-    ) util.Randomness
+    // Chain returns the associated chain which can be used to
+    // derive randomness in this runtime
+    Chain()            chain.Chain
```

Comment:
No no absolutely not! The runtime must abstract away the chain and provide a high level interface to what the caller actually needs. Actors do not need access to the full chain.

Reply:
I don't see how we can get access to the appropriate randomness at various points in the chain without passing in the chain (an abstraction atop a tipset basically)... This is how lotus does it if I'm not mistaken.